### ISSUE TITLE
When the machine is unplugged, the session.xml file will be corrupted.

### ISSUE THREAD

**Issue by expwnent** [notepad-plus-plus/notepad-plus-plus#5305]
### Description of the Issue
When the machine running notepad++ is unplugged, the previous session will fail to restore after the machine is rebooted.

### Steps to Reproduce the Issue
1. Have notepad++ open and set to periodically back up the current session.
2. Unplug your machine.
3. Be unlucky.
4. Power on your machine.
5. Open notepad++ and observe that your session is not restored.

### Expected Behavior
When notepad++ is re-opened, the previous session should be restored.

### Actual Behavior
The previous session is corrupted, in my most recent case in the form of 36 thousand null characters instead of a properly-formatted xml file.

It also tends to lose certain settings, like whether auto-indent is enabled in each file type. Even manual backups do not seem to restore this kind of setting, probably because the files that contain them are corrupted simultaneously to session backups being corrupted. They are probably corrupted for the same reason.

### Debug Information
Notepad++ v7.5.9   (64-bit)
Build time : Oct 14 2018 - 15:19:55
Path : C:\Programs\Notepad++\notepad++.exe
Admin mode : ON
Local Conf mode : OFF
OS : Windows 10 (64-bit)
Plugins : DSpellCheck.dll mimeTools.dll NppConverter.dll 

Note that manual backups of the current session do still work correctly, but it is inconvenient to need to remember to make manual backups.

I cannot directly reproduce the problem but I'm perfectly willing to use a special "debug" version of the program which will capture information to help find the cause the next time it happens. It typically happens to me a few times a year so it's likely it will not stop.

The most likely explanation is that notepad++ notices that it is closing, deletes the old session.xml file (in the relevant directory (the same as the install directory in my case)), starts writing the current session to session.xml, and the machine is fully shut down before the file can be written. The easy solution to this would be to write the session to session.xml.tmp and only delete the old session.xml after verifying that the new backup is properly formatted.

**Comment by sasumner** [comment]
You are lucky you have only lost Notepad++ "meta" information.  Tons of people have lost all of their file contents as well. Typically when that happens their files are filled with NUL characters.

**Comment by expwnent** [comment]
I've lost files now and then too, but not in the most recent version, and only files that haven't been saved yet, that just say "new 4" or whatever in the title bar. Those are supposed to be backed up too, but they can be corrupted if you have to power off your machine.

**Comment by sasumner** [comment]
That NUL issue may be history now, supposed to be fixed in one of the later 7.5.x releases.

**Comment by MetaChuh** [comment]
please update your notepad++ to the latest version [<kbd>Notepad++ 7.6.6</kbd>](https://notepad-plus-plus.org/download/v7.6.6.html) and migrate or reinstall your plugins if you update from version 7.5.9 or below.
if your issue still persists, and it is still of relevance to you, please feel free to file a new issue as easy to understand and reproducible as possible.

thank you for your help and best regards.

-----
```
general notes:

please close your issues as soon as you are satisfied, to help
keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

please also conduct a search at the issue tracker beforehand, to check if a similar issue
already exists.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
(no extra account is needed, just use your github account to sign in)
```
[<p align="center"><kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd></p>](https://notepad-plus-plus.org/community/)
